### PR TITLE
Feature/contracts agreements customerio

### PR DESCRIPTION
**Pull request by johantjellden**
A bit left to do but please check that I use the correct events.

### CODE REVIEW

**Comment by johantjellden** [comment]
Should be feature complete now.

File: src/main/java/com/hedvig/contract/customerio/EventListener.kt [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    replayStatus: ReplayStatus
+  ) {
+    if(replayStatus == ReplayStatus.REPLAY)
+      return
+
+    sendCancellation(event, timestamp, event.terminationDate, event.contractId)
+  }
+
+  @EventHandler
+  fun onContractWithPendingAgreementTerminated(event: ContractTerminated, @Timestamp timestamp:Instant,
```
Comment (meletios) [inline src/main/java/com/hedvig/contract/customerio/EventListener.kt]: Change name of method

File: src/main/java/com/hedvig/contract/customerio/EventListener.kt [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+@Order(1)
+class EventListener(
+  val customerio:CustomerioClient,
+  val contractStateRepository: ContractStateRepository<AgreementState, ContractState<AgreementState>>) {
+
+  val logger = LoggerFactory.getLogger(ContractStateProjection::class.java)!!
+
+  @EventHandler
+  fun onContractCreated(
+    event: ContractWithActiveAgreementCreated<AgreementStereotype>,
```
Comment (meletios) [inline src/main/java/com/hedvig/contract/customerio/EventListener.kt]: we need an event for pending agreement here

File: src/main/java/com/hedvig/contract/customerio/EventListener.kt [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  fun onContractWithPendingAgreementTerminationReverted(event: ContractWithPendingAgreementTerminationReverted,
+    replayStatus: ReplayStatus
+  ) {
+    if(replayStatus == ReplayStatus.REPLAY)
+      return
+    removeCancellationDate(event, event.contractId)
+  }
+
+  @EventHandler
+  fun onContractWithPendingAgreementTerminationReverted(event: ContractTerminationReverted,
```
Comment (meletios) [inline src/main/java/com/hedvig/contract/customerio/EventListener.kt]: Change name of the method here

File: src/main/java/com/hedvig/contract/customerio/EventListener.kt [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+    if (contract != null) {
+      val data = mapOf("renewal_date" to event.queueDate.toStockholmInstant().epochSecond)
+      customerio.updateCustomer(contract.holderMemberId, data)
+    } else {
+      logger.error("Could not find a contract for event $event")
+    }
+
+  }
+
```
Comment (meletios) [inline src/main/java/com/hedvig/contract/customerio/EventListener.kt]: We need also a termination date changed event

**Pull request merged**